### ISSUE TITLE
Add get_ref to result and adjust unit tests

### ISSUE THREAD

**Issue by lexicalunit** [nanodbc/nanodbc#52]
_From @ctrychta on July 17, 2014 16:13_

I found that when processing large result sets that a lot of time was being spent creating new strings so I added a get_ref function which will fill in the reference with the column value instead of returning it.

While adding unit tests for these changes I ran into a few problems with the existing tests.  I'm not sure how the tests were previously passing so if you think this could somehow be related to something in my environment please let me know.  After these changes I was able to successfully run the sqlite tests on linux and windows.

_Copied from original issue: lexicalunit/nanodbc/pull/25_

**Comment by lexicalunit** [comment]
Regarding the commit message: "Changed the check for affected_rows in simple_test to be 0 instead of 4.
This was needed because the SQLite driver always returns 0 for select
statements." That's ddd, because my driver is returning 4 affected rows for the select statement. I wonder if there's some setting in ODBC that affects this or ... what?

Regardless, I fixed-up f0e8d73d60ca5cb49dcc0c93a31def66233f76d4 with a small comment  and change to the test indicating that some drivers will return the number of rows selected, and others will return 0. I also amended some additional tabs to spaces conversions to c44be452db08d5099f9cf4556b824471fbf2153d, good catch!

This is a very useful, well written, and organized changeset, thank you!

**Comment by lexicalunit** [comment]
_From @ctrychta on July 17, 2014 17:45_

From the sqliteodbc source that I compiled I see the following implementation of SQLRowCount:

```
SQLRowCount(SQLHSTMT stmt, SQLLEN *nrows)
{
    STMT *s;

    HSTMT_LOCK(stmt);
    if (stmt == SQL_NULL_HSTMT) {
    return SQL_INVALID_HANDLE;
    }
    s = (STMT *) stmt;
    if (nrows) {
    *nrows = s->isselect ? 0 : s->nrows;
    }
    HSTMT_UNLOCK(stmt);
    return SQL_SUCCESS;
}
```

which makes it seem like select would always result in a zero row count so I'm not sure why we would be seeing different results.

Regardless, thanks for merging so quickly.
